compiler/rustc_resolve/src/rustdoc.rs

```diff
@@ -439,13 +441,25 @@ fn parse_links<'md>(doc: &'md str) -> Vec<Box<str>> {
                         links.push(display_text);
                     }
                 }
+                if matches!(
+                    link_type,
+                    LinkType::Reference | LinkType::Shortcut | LinkType::Collapsed
+                ) {
+                    refids.insert(id);
+                }
 
                 links.push(preprocess_link(&dest_url));
             }
             _ => {}
         }
     }
 
+    for (label, refdef) in event_iter.reference_definitions().iter() {
```

Comment:
This logic is run on a compiler's happy path, when rustdoc and its lints are not involved.
Could you do a perf run before merging this?